From 2e9c6c9a75e5602cf30a30898e3771795d29d6d6 Mon Sep 17 00:00:00 2001
From: Jiamin Miao <jiamin.miao@amlogic.com>
Date: Thu, 14 May 2015 09:24:05 +0800
Subject: [PATCH 5624/5965] PD#106566: uart: use clk81 to support high baud
 rate.

In XTAL clock source, the data is not stable when baud > 2M b/s.

To support > 2M baud rate, uart clock should

change to use clk81. so only uart A0 use xtal as

clock source, the other uarts all change to use clk81

Change-Id: Ibf9ab4c729d4cb7631e67c302c95e816bf19a44b
---
 drivers/amlogic/uart/uart/meson_uart.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/uart/uart/meson_uart.c b/drivers/amlogic/uart/uart/meson_uart.c
index c7e861624dfa..c5a9a8976101 100644
--- a/drivers/amlogic/uart/uart/meson_uart.c
+++ b/drivers/amlogic/uart/uart/meson_uart.c
@@ -353,7 +353,7 @@ static void meson_uart_change_speed(struct meson_uart_port *mup, unsigned long n
 	if (!newbaud || newbaud == mup->baud)
 		return;
 
-	if (IS_MESON_M8B_CPU || IS_MESON_M8M2_CPU) {
+	if ((IS_MESON_M8B_CPU || IS_MESON_M8M2_CPU) && (0 == mup->line)) {
 		msleep(1);
 		xtal = clk_get_sys("xtal", NULL);
 		value = (clk_get_rate(xtal) / 3) / newbaud - 1;
@@ -382,7 +382,7 @@ static void meson_uart_change_speed(struct meson_uart_port *mup, unsigned long n
 	value |= 0x800000;
 
 	//Set USE_XTAL_CLK bit
-	if(IS_MESON_M8B_CPU || IS_MESON_M8M2_CPU)
+	if ((IS_MESON_M8B_CPU || IS_MESON_M8M2_CPU) && (0 == mup->line))
 		value  |= 1<<24;
 
 	writel(value, &uart->reg5);
-- 
2.19.0

